src/analysis/functions.rs

```diff
@@ -14,21 +15,27 @@ pub struct Info {
     pub ret: Option<library::Parameter>,
 }
 
-pub fn analyze(env: &Env, type_: &library::Class, class_tid: library::TypeId) -> Vec<Info> {
+pub fn analyze(env: &Env, type_: &library::Class, class_tid: library::TypeId,
+    used_types: &mut HashSet<String>) -> Vec<Info> {
     let mut funcs = Vec::new();
 
     for func in &type_.functions {
-        let info = analyze_function(env, func, class_tid);
+        let info = analyze_function(env, func, class_tid, used_types);
         funcs.push(info);
     }
 
     funcs
 }
 
-fn analyze_function(env: &Env, type_: &library::Function, class_tid: library::TypeId) -> Info {
+fn analyze_function(env: &Env, type_: &library::Function, class_tid: library::TypeId,
+    used_types: &mut HashSet<String>) -> Info {
     let mut commented = false;
 
-    let ret = if type_.ret.typ == Default::default() { None } else { Some(type_.ret.clone()) };
+    let ret = if type_.ret.typ == Default::default() { None } else {
+        let _ = used_rust_type(&env.library, type_.ret.typ)
+            .map(|s| used_types.insert(s));
```

Comment:
`Result::ok` seems a good fit for this: `used_rust_type(&env.library, type_.ret.typ).ok().map(|s| used_types.insert(s));`